pkg/queryfrontend/roundtrip.go

```diff
@@ -50,10 +55,12 @@ func NewTripperWare(
 		queryrange.StepAlignMiddleware,
 	)
 
-	if splitQueryInterval != 0 {
+	codec := NewThanosCodec(config.PartialResponseStrategy)
+
+	if config.SplitQueriesByInterval != 0 {
```

Comment:
Is this still necessary? It is already checked in https://github.com/thanos-io/thanos/pull/3114/files#diff-98bce32adf2d7692d381e0688ccaab15R137